escrow/test/Escrow.js

```diff
@@ -0,0 +1,106 @@
+    await expect(escrowInstance.connect(requestor).serviceProviderConfirmTaskCompletion(true)).to.
+      be.revertedWith("Only the service provider can confirm his part")
+  });
+
+  it("should let the both parties confirm the task completion and send tokens to the service provider", async function () {
+    // transfer funds to the contract
+    const previousContractBalance = await acaInstance.balanceOf(escrowInstance.address);
+    await acaInstance.transfer(escrowInstance.address, ESCROW_AMOUNT, { from: requestorAddress })
+    const currentContractBalance = await acaInstance.balanceOf(escrowInstance.address);
+    expect(currentContractBalance).to.equal(previousContractBalance.add(ESCROW_AMOUNT));
+
+    // requestor and service provider confirm completion
+    await escrowInstance.connect(provider).serviceProviderConfirmTaskCompletion(true)
+    const initialProviderNativeBalance = await provider.getBalance();
+    const initialProviderBalance = await acaInstance.balanceOf(providerAddress);
+    await escrowInstance.connect(requestor).requestorConfirmTaskCompletion(true)
+
+    // complete the task and check if the provider balance increased
+    const finalProviderBalance = await acaInstance.balanceOf(providerAddress);
+    expect(finalProviderBalance).to.equal(initialProviderBalance.add(ESCROW_AMOUNT));
+
+    // the change in native ACA balance should match the change in ERC20 ACA balance
+    const finalProviderNativeBalance = await provider.getBalance();
+    // multiplied the escrow amount by 10⁶ because there's a difference in
+    // BigNumber's decimals between native and ERC20 balance
+    const expectedProviderNativeBalance = initialProviderNativeBalance.add(ESCROW_AMOUNT * 1_000_000);
+    expect(finalProviderNativeBalance).to.equal(expectedProviderNativeBalance);
+  });
+
+  it("should let the both parties deny the task completion and send tokens back to the requestor", async function () {
+    // transfer funds to the contract
+    const previousContractBalance = await acaInstance.balanceOf(escrowInstance.address);
+    await acaInstance.transfer(escrowInstance.address, ESCROW_AMOUNT, { from: requestorAddress })
+    const currentContractBalance = await acaInstance.balanceOf(escrowInstance.address);
+    expect(currentContractBalance).to.equal(previousContractBalance.add(ESCROW_AMOUNT));
+
+    // requestor and service provider deny completion
+    await escrowInstance.connect(requestor).requestorConfirmTaskCompletion(false);
+    const initialRequestorBalance = await acaInstance.balanceOf(requestorAddress);
+    await escrowInstance.connect(provider).serviceProviderConfirmTaskCompletion(false);
+
+    // complete the task and check if the requestor balance increased
+    const finalRequestorBalance = await acaInstance.balanceOf(requestorAddress);
+    expect(finalRequestorBalance).to.equal(initialRequestorBalance.add(ESCROW_AMOUNT));
+  });
+});
```

Comment:
I just realized the other tutorials break the test in different `describes` like `Deployment` and `Operation`. Should we do this here too?

Reply:
Yes, if you figure out more descriptive or granual section than `Deployment` and `Operation`, I encourage you to use them.